great_expectations/datasource/util.py

```diff
@@ -2,16 +2,45 @@
 
 logger = logging.getLogger(__name__)
 
-try:
-    from pyspark import SparkContext
-    from pyspark.sql import SparkSession
 
-    from great_expectations.datasource.sparkdf_datasource import SparkDFDatasource
-except ImportError:
-    SparkContext = None
-    SparkSession = None
-    SparkDFDatasource = None
-    # TODO: review logging more detail here
-    logger.debug(
-        "Unable to load pyspark; install optional spark dependency for support."
+# noinspection PyPep8Naming
+def get_or_create_spark_session(
+    name: str = "default_great_expectations_spark_dataframe_datasource",
```

Comment:
Why would we default to changing app.name?

Reply:
@jcampbell This kicks in when the user DOES NOT specify spark_config -- i.e., the `len(spark_config) == 0` case.